src/views/Users/User.vue

```diff
@@ -212,50 +212,49 @@ export default {
             return this.$SystemAPI.userMembershipRemove({ roleID, userID })
           }
         }
-      })).then(() => {
-        this.fetchUserRoles()
-      })
+      }))
+        .then(this.fetchUserRoles)
+        .catch(this.stdReject)
+        .finally(this.finalize)
     },
 
     onStatusChange () {
       this.processing = true
       const userID = this.userID
+      let endpoint
       if (this.user.suspendedAt) {
-        this.$SystemAPI.userUnsuspend({ userID })
-          .catch(this.stdReject)
-          .finally(() => {
-            this.fetchUser()
-          })
+        endpoint = this.$SystemAPI.userSuspend
       } else {
-        this.$SystemAPI.userSuspend({ userID })
-          .catch(this.stdReject)
-          .finally(() => {
-            this.fetchUser()
-          })
+        endpoint = this.$SystemAPI.userUnsuspend
       }
+
+      endpoint({ userID })
+        .then(() => {
+          this.fetchUser()
+        })
+        .catch(this.stdReject)
+        .finally(this.finalize)
     },
 
     onPasswordSubmit () {
       this.processing = true
       const { userID, password } = this.user
       this.$SystemAPI.userSetPassword({ userID, password })
-        .catch(this.stdReject)
-        .finally(() => {
+        .then(() => {
```

Comment:
@darh I did this change so the password fields get cleared only after a successful request. It would be annoying if they get cleared if something goes wrong and you have to retype it.